src/Driver/Driver.php

```diff
@@ -605,13 +605,7 @@ protected function bindParameters(PDOStatement $statement, iterable $parameters)
      */
     protected function formatDatetime(DateTimeInterface $value): string
     {
-        try {
-            $datetime = new DateTimeImmutable('now', $this->getTimezone());
-        } catch (Throwable $e) {
-            throw new DriverException($e->getMessage(), $e->getCode(), $e);
-        }
-
-        return $datetime->setTimestamp($value->getTimestamp())->format(static::DATETIME);
+        return $value->setTimezone($this->getTimezone())->format(static::DATETIME);
```

Comment:
In this case `$value` can be changed if it is instance of mutable class (`DataTime`).
I think it will better to clone `$value` and then change it:
```suggestion
        return (clone $value)->setTimezone($this->getTimezone())->format(static::DATETIME);
```

Reply:
Agreed considering the original function made no attempts at modifying the original date time instance but would have still preferred the function have that option and the immutability managed by whether an instance of DateTime or DateTimeImmutable be passed.

Reply:
> the immutability managed by whether an instance of DateTime or DateTimeImmutable be passed.

This will affect backward compatibility.